packages/react/src/hooks/useMerged.js

```diff
@@ -0,0 +1,8 @@
+import { useMemo } from 'react';
+import { merge } from 'lodash-es';
+
+const useMerged = (...args) => {
+  return useMemo(() => merge({}, ...args), [args]);
```

Comment:
I could be wrong, but I don't think this doing what you think it's doing.

you are passing in an arg like `useMeged(obj1, obj2)` and you are spreading them for the merge, which is fine, but the useMemo(, [args]) is getting a new array instance every time, and likely it means that every place we are using useMerge we doing extra work to merge and memorize it but nothing is ever memorized.   (ie, if I pass in (obj1, obj2) then what I'm passing to the hook in the array is [[obj1, obj2]] and not [obj1, obj2].

it could be that `return useMemo(() => merge({}, ...args), args);` might work, but you'd need to unit test it.  Or maybe... `return useMemo(() => merge({}, ...args), [...args]);` but again, you'd have to create a unit test to verify this.

As I said, I could be wrong, and maybe the hook is smart enough to handle this, but I'm skeptical.  Could you confirm.

Reply:
@stuckless You are absolutely correct. Good catch. Because of the de-structuring here we are creating a new variable each time this is called. @kevinsperrine We can fix this by using a regular function and spreading the `arguments` variable. I have demo'd [here](https://codesandbox.io/s/fast-hill-jtu61).

Reply:
Nice touch using the "old" arguments object, haven't seen that one in a while